### PR TITLE
Add UT for workstatus.go and taint.go and rule.go

### PR DESCRIPTION
**Pull request by realnumber666**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Promote UT coverage

**Which issue(s) this PR fixes**:
Part of #3327 

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

File: pkg/util/helper/taint_test.go [hunk 1]
```diff
@@ -357,3 +385,132 @@ func TestGetNoExecuteTaints(t *testing.T) {
 		})
 	}
 }
+
+func TestGetMinTolerationTime(t *testing.T) {
```
Comment (XiShanYongYe-Chang) [inline pkg/util/helper/taint_test.go:389]: Some paths may not be tested. Can you add more test items?

image

File: pkg/util/helper/taint_test.go [hunk 2]
```diff
@@ -357,3 +385,132 @@ func TestGetNoExecuteTaints(t *testing.T) {
...
+		{
+			Key:    "key2",
+			Value:  "value2",
+			Effect: corev1.TaintEffectNoSchedule,
+		},
+	}
+	expected := true
+	expectedTolerations := tolerations
+	actual, actualTolerations := GetMatchingTolerations(taints, tolerations)
+	if actual != expected {
```
Comment (XiShanYongYe-Chang) [inline pkg/util/helper/taint_test.go:472]: Here can simply `expected` to true.

In addition, can more test items be added to cover the test path?

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks

File: pkg/util/helper/taint_test.go (hunk 1 above)
Comment (realnumber666) [reply pkg/util/helper/taint_test.go:389]: ok

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks~
/lgtm

Ask @RainbowMango to help rerun the test.

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**